postgres_composite_types/forms.py

```diff
@@ -32,15 +32,47 @@
 Takes inspiration from django.forms.MultiValueField/MultiWidget.
 """
 
+import copy
 import logging
 from collections import OrderedDict
 
 from django import forms
 from django.utils.safestring import mark_safe
 
+from . import CompositeType
+
 LOGGER = logging.getLogger(__name__)
 
 
+class CompositeBoundField(forms.BoundField):
+    """
+    Allow access to nested BoundFields for fields. Useful for customising the
+    rendering of a CompositeTypeField:
+
+        <label for="{{ form.address.id_for_widget }}">Address:</label>
+        {{ form.address.address_1 }}
+        {{ form.address.address_2 }}
+        <label for="{{ form.address.suburb }}">Suburb:</label>
+        {{ form.address.suburb }}
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._bound_fields_cache = {}
+
+        initial = self.form.initial.get(self.name, self.field.initial)
+        if isinstance(initial, CompositeType):
+            initial = initial.__to_dict__()
+        data = self.form.data if self.form.is_bound else None
```

Comment:
This one is required, sadly. `form.data` is set to `{}` if `None` is passed in, but passing `{}` means a very different thing to passing in `None`. I can make it an explicit `if: else:` block if you prefer.